x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -34,31 +43,60 @@ export const validateEql = async ({
   signal,
   runtimeMappings,
   options,
-}: Params): Promise<{ valid: boolean; errors: string[] }> => {
-  const { rawResponse: response } = await firstValueFrom(
-    data.search.search<EqlSearchStrategyRequest, EqlSearchStrategyResponse>(
-      {
-        params: {
-          index: dataViewTitle,
-          body: { query, runtime_mappings: runtimeMappings, size: 0 },
-          timestamp_field: options?.timestampField,
-          tiebreaker_field: options?.tiebreakerField || undefined,
-          event_category_field: options?.eventCategoryField,
+}: Params): Promise<{
+  valid: boolean;
+  error?: { code: EQL_ERROR_CODES; messages?: string[]; error?: Error };
+}> => {
+  try {
+    const { rawResponse: response } = await firstValueFrom(
+      data.search.search<EqlSearchStrategyRequest, EqlSearchStrategyResponse>(
+        {
+          params: {
+            index: dataViewTitle,
+            body: { query, runtime_mappings: runtimeMappings, size: 0 },
+            timestamp_field: options?.timestampField,
+            tiebreaker_field: options?.tiebreakerField || undefined,
+            event_category_field: options?.eventCategoryField,
+          },
+          options: { ignore: [400] },
         },
-        options: { ignore: [400] },
+        {
+          strategy: EQL_SEARCH_STRATEGY,
+          abortSignal: signal,
+        }
+      )
+    );
+    if (isParsingErrorResponse(response)) {
```

Comment:
Since this function is growing in complexity, it's probably time to add some unit tests around it (more for documentation of behavior than anything else); maybe extract a function that receives a response and test that? That would also allow us to document the shape of response(s) we handle.

```ts
type EqlValidationTransformer = (response: unknown) => {
  valid: boolean;
  error?: { code: EQL_ERROR_CODES; messages?: string[]; error?: Error };
};
```

Reply:
Great point! I will add tests for this functionality